reactive_graph/src/effect/effect.rs

```diff
@@ -196,6 +198,57 @@ impl Effect<LocalStorage> {
         Self { inner }
     }
 
+    /// Creates a new effect, which runs once on the next “tick”, and then runs again when reactive values
+    /// that are read inside it change.
+    ///
+    /// This spawns a task on the local thread using
+    /// [`spawn_local`](any_spawner::Executor::spawn_local). For an effect that can be spawned on
+    /// any thread, use [`new_sync`](Effect::new_sync).
+    pub fn new_with_runtime<T, M>(
+        mut fun: impl EffectFunction<T, M> + 'static,
+        pass_to_rt: impl FnOnce(Pin<Box<dyn Future<Output = ()> + 'static>>),
```

Comment:
I'd suggest a name like `spawn` here rather than `pass_to_rt` as that's the phrase we tend to use for "spawn a task" elsewhere

I think the signature can be something like
```
    pub fn new_with_runtime<T, M, F, Fut>(
        mut fun: impl EffectFunction<T, M> + 'static,
        pass_to_rt: F
    ) -> Self 
    where 
        T: 'static,
        F: FnOnce(Fut) + 'static,
        Fut: Future<Output = ()>
```
if you want to avoid the `Pin<Box<_>>` here